cloudinary-core/src/main/java/com/cloudinary/Api.java

```diff
@@ -552,4 +560,81 @@ private ApiResponse updateResourcesAccessMode(String accessMode, String byKey, O
         return callApi(HttpMethod.POST, uri, params, options);
     }
 
+    /**
+     * Add a new metadata field definition
+     * @param field The field to add.
+     * @return A map representing the newlay added field.
+     * @throws Exception
+     */
+    public ApiResponse addMetadataField(AbstractMetadataField field) throws Exception {
+        return callApi(HttpMethod.POST, Collections.singletonList("metadata_fields"),
+                ObjectUtils.toMap(field), ObjectUtils.asMap ("content_type", "json"));
+    }
+
+    /**
+     * List all the metadata field definitions (structure, not values)
+     * @return A map containing the list of field definitions.
```

Comment:
@nitzanj , when you say "field definition", you mean AbstractMetadataField object? or some json?

Reply:
It is returned as maps, I added it to the doc string.